src/core/control/Control.cpp

```diff
@@ -1084,6 +1087,37 @@ void Control::selectFillAlpha(bool pen) {
     }
 }
 
+void Control::selectAndEndText() {
+    auto xournal = this->win->getXournal();
```

Comment:
This seems to duplicate existing functionality. I think the logic can be cleaned up as follows:

* First, check whether there is a text editor, and whether the text is currently non-empty. If not, then return. (There is nothing to be done here).
* Save the pointer to the text element.
* Call `clearSelectionEndText()` to close the text editor. Note that if the text element was actually added, it should be on some layer on some page (see `XojPageView::endText()` in `PageView.cpp`).
* If the pointer to the text element is not null, then set the current selection to the text element.

To further clean this up, we can add the type of the selection tool as a function parameter and then move the `selectTool()` logic in `actionPerformed()` into this method, so that all we need to do is call this method with the right tool type.

If you want to follow my suggestion here, I would also suggest renaming this method to something like `handleSelectToolAction`.

Reply:
I'm not sure if this would actually clean up code.

Reply:
Thank you very much. I think this really cleans up a lot and works just as fine.
Although I'm not sure about moving the `selectTool()` logic in `actionPerformed()` into this method. It only requires another check if the selected tool is one of the selection tools every time it is called.
I still implemented it, but would recommend checking on this again.